### PR TITLE
Changes identifier to zero or more for publications.

### LINKED ISSUE #149: Make identifiers zero or more for publication
Since not all publications have identifiers.

refs https://github.com/sul-dlss/rialto-etl/issues/274

### PR DESCRIPTION
**Pull request by justinlittman**
closes #149

### CODE REVIEW

File: transform/publication_indexer.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ func (m *PublicationIndexer) Index(resource *models.Publication, doc solr.Docume
 	doc.Set("type_ssi", "Publication")
 	doc.Set("title_tesi", resource.Title)
 	doc.Set("created_ssim", resource.Created)
-	doc.Set("identifier_ssim", resource.Identifier)
+	// doc.Set("identifier_ssim", resource.Identifier)
```
Comment (jcoyne) [inline transform/publication_indexer.go]: Cruft?

File: transform/publication_indexer.go [hunk 2]
```diff
@@ -26,6 +26,8 @@ func (m *PublicationIndexer) Index(resource *models.Publication, doc solr.Docume
 	m.indexAuthors(resource, &doc)
 	m.indexConcepts(resource, &doc)
 	m.indexGrants(resource, &doc)
+	// m.indexIdentifiers(resource, &doc)
```
Comment (jcoyne) [inline transform/publication_indexer.go]: cruft?

**Comment by justinlittman** [comment]
@jcoyne Decrufting pushed.

File: models/publication_test.go [hunk 3]
```diff
@@ -12,12 +12,10 @@ func TestNewPublicationMinimalFields(t *testing.T) {
 	id, _ := rdf.NewIRI("http://example.com/record1")
 	title, _ := rdf.NewLiteral("Hello world!")
 	created, _ := rdf.NewLiteral("2004-06-11?")
-	identifier, _ := rdf.NewLiteral("1234567")
```
Comment (mjgiarlo) [inline models/publication_test.go]: Would you consider adding coverage of the changes you made to the pub model to this test, or do you think this is sufficiently covered already?

Comment (justinlittman) [reply models/publication_test.go]: So we don't have tests for any of the SetXXInfo() methods for any of the model entities. It's fairly boilerplate code, but definitely testable.

I defer to your recommendation on whether tests should be added for these methods.

Comment (mjgiarlo) [reply models/publication_test.go]: Meh, you convinced me not to care. :)

**Pull request merged**